### PR TITLE
[NTUSER] menu.c: Don't use RETURN() macro

### PR DESCRIPTION
**Pull request by TAN-Gaming**
## Purposed changes
- Don't use RETURN() macro
- Remove parentheses around return value

### CODE REVIEW

File: win32ss/user/ntuser/menu.c [hunk 1]
```diff
@@ -6041,16 +6030,17 @@ NtUserGetMenuBarInfo(
...
-      RETURN(FALSE);
+      Ret = FALSE;
+      goto Cleanup;
    }
 
-   RETURN(TRUE);
+   Ret = TRUE;
 
-CLEANUP:
+Cleanup:
```
Comment (JoachimHenze) [inline win32ss/user/ntuser/menu.c:6039]: In some cases you renamed the Cleanup label to Exit, and in some cases you decided to stick with the name Cleanup. Is that difference intended anyhow? Or would it be better to structurally use the same name, e.g. Exit?

File: win32ss/user/ntuser/menu.c [hunk 2]
```diff
@@ -6064,27 +6054,29 @@ NtUserGetMenuIndex(
...
+      goto Exit;
 
    MenuItem = Menu->rgItems;
    for (i = 0; i < Menu->cItems; i++, MenuItem++)
    {
        if (MenuItem->spSubMenu == SubMenu)
-          RETURN(MenuItem->wID);
+       {
+          Ret = MenuItem->wID;
+          break;
```
Comment (JoachimHenze) [inline win32ss/user/ntuser/menu.c:6072]:
```suggestion
          goto Exit;
```
May be more safe if anything "dangerous" would ever be placed after the for-loop.

File: win32ss/user/ntuser/menu.c [hunk 3]
```diff
@@ -6581,23 +6562,27 @@ NtUserThunkedMenuItemInfo(
...
+   {
+      Ret = UserInsertMenuItem(Menu, uItem, fByPosition, lpmii, &lstrCaption);
+      goto Cleanup;
+   }
 
-   RETURN( UserMenuItemInfo(Menu, uItem, fByPosition, (PROSMENUITEMINFO)lpmii, TRUE, &lstrCaption));
+   Ret = UserMenuItemInfo(Menu, uItem, fByPosition, (PROSMENUITEMINFO)lpmii, TRUE, &lstrCaption);
 
-CLEANUP:
+Cleanup:
```
Comment (JoachimHenze) [inline win32ss/user/ntuser/menu.c:6577]: Same here. Maybe could be called Exit as well?

File: win32ss/user/ntuser/menu.c (hunk 1 above)
Comment (TAN-Gaming) [reply win32ss/user/ntuser/menu.c:6039]: Cleanup is used when there is something to clean up other than `UserLeave`.

Comment (JoachimHenze) [reply win32ss/user/ntuser/menu.c:6039]: Okay, I almost anticipated that. And I am fine with that differentiation. 2 of my comments Resolved then.

File: win32ss/user/ntuser/menu.c (hunk 3 above)
Comment (JoachimHenze) [reply win32ss/user/ntuser/menu.c:6577]: Resolved. See above.

File: win32ss/user/ntuser/menu.c [hunk 4]
```diff
@@ -6183,29 +6175,29 @@ NtUserHiliteMenuItem(
 {
    PMENU Menu;
    PWND Window;
-   DECLARE_RETURN(BOOLEAN);
+   BOOL Ret = FALSE;
```
Comment (JoachimHenze) [inline win32ss/user/ntuser/menu.c:6178]: your reason for changing the type was...

File: win32ss/user/ntuser/menu.c (hunk 2 above)
Comment (TAN-Gaming) [reply win32ss/user/ntuser/menu.c:6072]: I don't think there will be a problem if we want to add more code there in the future. We can check the `Ret` value before continue.

Comment (JoachimHenze) [reply win32ss/user/ntuser/menu.c:6072]: Ok, not a biggy. Can live with that decision. Resolved.

File: win32ss/user/ntuser/menu.c (hunk 4 above)
Comment (TAN-Gaming) [reply win32ss/user/ntuser/menu.c:6178]: The correct return type of this function is BOOL.

Comment (JoachimHenze) [reply win32ss/user/ntuser/menu.c:6178]: Correct. Resolved and approved.

**Review by JoachimHenze — APPROVED** [review]

**Comment by JoachimHenze** [comment]
I am not attaching these test-results to torture or nag you, it is because I am interested in your changes, and want to increase the level of trust in your changes:
KVM https://reactos.org/testman/compare.php?ids=92086,92089 LGTM
VBox https://reactos.org/testman/compare.php?ids=92085,92091 (does show 2 additional reboots)
The bots are known to be in a poor state currently, the bad result is not (necessarily/likely) related to this PR.
I do see no strong relation to what has been committed here. Needs more tests.

**Pull request merged**